src/webgpu/api/operation/render_pipeline/culling_tests.spec.ts

```diff
@@ -0,0 +1,151 @@
+`;
+
+import { poptions, params } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+function faceIsCulled(face: 'cw' | 'ccw', frontFace: GPUFrontFace, cullMode: GPUCullMode): boolean {
+  return cullMode !== 'none' && (frontFace === face) === (cullMode === 'front');
+}
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('culling')
+  .params(
+    params()
+      .combine(poptions('frontFace', ['ccw', 'cw'] as const))
+      .combine(poptions('cullMode', ['none', 'front', 'back'] as const))
+      .combine(
+        poptions('depthStencilFormat', [
+          null,
+          'depth24plus',
+          'depth32float',
+          'depth24plus-stencil8',
+        ] as const)
+      )
+      .combine(poptions('primitiveTopology', ['triangle-list'] as const))
+  )
+  .fn(t => {
+    const size = 4;
+    const format = 'rgba8unorm';
+
+    const texture = t.device.createTexture({
+      size: { width: size, height: size, depth: 1 },
+      format,
+      usage: GPUTextureUsage.OUTPUT_ATTACHMENT | GPUTextureUsage.COPY_SRC,
+    });
+
+    const depthTexture = t.params.depthStencilFormat
+      ? t.device.createTexture({
+          size: { width: size, height: size, depth: 1 },
+          format: t.params.depthStencilFormat,
+          usage: GPUTextureUsage.OUTPUT_ATTACHMENT,
+        })
+      : null;
+
+    const encoder = t.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: texture.createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 1.0, a: 1.0 },
+        },
+      ],
+      depthStencilAttachment: depthTexture
+        ? {
+            attachment: depthTexture.createView(),
+            depthLoadValue: 1.0,
+            depthStoreOp: 'store',
+            stencilLoadValue: 0,
+            stencilStoreOp: 'store',
+          }
+        : undefined,
+    });
+
+    // Draw two triangles with different winding orders:
+    // 1. The top-left one is counterclockwise (CCW)
+    // 2. The bottom-right one is clockwise (CW)
+    const vertexModule = t.makeShaderModule('vertex', {
+      glsl: `#version 450
+            const vec2 pos[6] = vec2[6](vec2(-1.0f,  1.0f),
+                                        vec2(-1.0f,  0.0f),
+                                        vec2( 0.0f,  1.0f),
+                                        vec2( 0.0f, -1.0f),
+                                        vec2( 1.0f,  0.0f),
+                                        vec2( 1.0f, -1.0f));
+            void main() {
+                gl_Position = vec4(pos[gl_VertexIndex], 0.0, 1.0);
+            }`,
+    });
+
+    const fragmentModule = t.makeShaderModule('fragment', {
+      glsl: `#version 450
+      layout(location = 0) out vec4 fragColor;
+      void main() {
+        fragColor = vec4(1.0, 0.0, 0.0, 1.0);
```

Comment:
> We should also have a similar test (or even the same one) for the gl_FrontFace builtin.

We should be able to do this easily now, just make this output red or green depending on the gl_FrontFace value.